File: src/example-publish.py
#!venv/bin/python
import os, dotenv, autobahn.asyncio.component, asyncio

DEFAULT_ENV = '.env.service';

if os.path.exists(DEFAULT_ENV):
    dotenv.load_dotenv(DEFAULT_ENV)

if (
    os.environ.get('WAMP_DEBUG') is None
    or os.environ.get('WAMP_HOST') is None
    or os.environ.get('WAMP_PORT') is None
    or os.environ.get('WAMP_AUTHID') is None
    or os.environ.get('WAMP_SECRET') is None
    or os.environ.get('WAMP_REALM') is None
): exit()

component = autobahn.asyncio.component.Component(
    transports='wss://'
    + os.environ.get('WAMP_HOST') + ':' + os.environ.get('WAMP_PORT'),
    realm=os.environ.get('WAMP_REALM'),
    authentication={
        'wampcra': {
            'authid': os.environ.get('WAMP_AUTHID'),
            'secret': os.environ.get('WAMP_SECRET'),
        }
    },
)


@component.on_join
async def joined(session, details):
    while True:
        message = [
            'This is py first argument.',
            'This is py second argument.'
        ]
        session.publish('test', *message)
        print('Sent:', message)
        await asyncio.sleep(1)


if __name__ == '__main__':
    autobahn.asyncio.component.run(
        [component],
        True,
        'trace' if os.environ.get('WAMP_DEBUG') == 'true' else 'critical'
    )

File: src/example-publish.js
#!/usr/bin/node
const DEFAULT_ENV = '.env.service';

if (require('fs').existsSync(DEFAULT_ENV)) {
    require('dotenv').config({ path: DEFAULT_ENV })
}

if (
    !process.env.WAMP_DEBUG
    || !process.env.WAMP_HOST
    || !process.env.WAMP_PORT
    || !process.env.WAMP_AUTHID
    || !process.env.WAMP_SECRET
    || !process.env.WAMP_REALM
) {
    process.exit();
}

AUTOBAHN_DEBUG = (process.env.WAMP_DEBUG == "true");

var autobahn = require('autobahn');
var connection = new autobahn.Connection({
    url: 'wss://'+ process.env.WAMP_HOST +':'+ process.env.WAMP_PORT,
    realm: process.env.WAMP_REALM,
    authmethods: ['wampcra'],
    authid: process.env.WAMP_AUTHID,
    onchallenge: function (session, method, extra) {
        if (method === 'wampcra') {
            return autobahn.auth_cra.sign(
                process.env.WAMP_SECRET,
                extra.challenge
            );
        }
    }
});

connection.onopen = function(session) {

    setInterval(function() {
        let message = [
            'This is js first argument.',
            'This is js second argument.'
        ];
        session.publish('test', message);
        console.log('Sent:', message);
    }, 1000);

};

connection.open();
